src/features/ai/sidebar/SelectionPanelSection.tsx

```diff
@@ -1,52 +1,151 @@
 import { useQuery } from '@tanstack/react-query';
 import { useAtomValue } from 'jotai';
+import { useCallback, useEffect, useState } from 'react';
+import { VscChevronLeft, VscChevronRight } from 'react-icons/vsc';
 import { useDebounce } from 'usehooks-ts';
 
-import { askForRewrite } from '../../../api/rewrite';
+import { askForRewrite, REWRITE_MANNER, RewriteOptions } from '../../../api/rewrite';
 import { selectionAtom } from '../../../atoms/selectionState';
 import { PanelSection } from '../../../components/PanelSection';
 import { emitEvent } from '../../../events';
+import { cx } from '../../../lib/cx';
+import { getCachedSetting } from '../../../settings/settingsManager';
 
 export function SelectionPanelSection() {
   const selection = useAtomValue(selectionAtom);
   const debouncedSelection = useDebounce(selection, 200);
 
+  return (
+    <PanelSection title="Selection">
+      {!selection && <p className="mb-1 mt-4 px-4 italic">Select some text in the editor to see it here.</p>}
+
+      {selection && <RewriteWidget className="mb-6" selection={debouncedSelection} />}
+    </PanelSection>
+  );
+}
+
+export function RewriteWidget({
+  selection,
+  className,
+  onChoiceSelected,
+}: {
+  selection: string;
+  className?: string;
+  onChoiceSelected?: (choice: string) => void;
+}) {
+  const openAiSettings = getCachedSetting('openAI');
+
+  const [rewriteOptions, setRewriteOptions] = useState<RewriteOptions>({
+    nChoices: 3,
+    manner: openAiSettings.manner,
+    temperature: openAiSettings.temperature,
+  });
+
+  useEffect(() => {
+    setRewriteOptions((state) => ({
+      ...state,
+      manner: openAiSettings.manner,
+      temperature: openAiSettings.temperature,
+    }));
+  }, [openAiSettings.manner, openAiSettings.temperature]);
+
   const {
     error,
     data: rewrite,
     refetch,
     isFetching,
   } = useQuery({
-    queryKey: [debouncedSelection], // clear rewrite when the selection changes
+    queryKey: [selection, rewriteOptions], // clear rewrite when the selection changes
     enabled: false,
-    queryFn: () => askForRewrite(debouncedSelection),
+    queryFn: () => askForRewrite(selection, rewriteOptions),
   });
 
-  return (
-    <PanelSection title="Selection">
-      <p className="my-4 px-4 italic">Select some text in the editor to see it here.</p>
+  const [selectedChoiceIndex, setSelectedChoiceIndex] = useState(0);
+  const [selectedChoice, setSelectedChoice] = useState('');
+
+  useEffect(() => {
+    if (rewrite && selectedChoiceIndex >= 0 && selectedChoiceIndex < rewrite.length) {
+      setSelectedChoice(rewrite[selectedChoiceIndex]);
+    }
+  }, [rewrite, selectedChoiceIndex]);
```

Comment:
I'm not sure having a state for selectedChoice is that useful. You could reset the index when rewrite changes:
```typescript
useEffect(() => {
    setSelectedChoiceIndex(0);
  }, [rewrite]);
```
and then display `rewrite[selectedChoiceIndex]` in the component

Reply:
agree that this can be improved.